File: frontend/api/sedi.py
"""Funzioni API per le sedi."""

from frontend.api.client import get, post


def get_sedi() -> list:
    """Restituisce tutte le sedi attive."""
    return get("/sedi/") or []


def get_sede(sede_id: int) -> dict | None:
    """Restituisce una sede per ID."""
    return get(f"/sedi/{sede_id}")


def crea_sede(nome: str, indirizzo: str, citta: str, capienza: int) -> dict | None:
    """Crea una nuova sede."""
    return post("/sedi/", data={
        "nome": nome,
        "indirizzo": indirizzo,
        "citta": citta,
        "capienza_massima": capienza,
    })

File: frontend/src/api/sedi.js
// ─────────────────────────────────────────────────────────────────────────────
// API — Sedi
// Schema: { id, nome, indirizzo, citta, capienza_massima }
// ─────────────────────────────────────────────────────────────────────────────

import { apiGet, apiPost, apiPatch } from './client'

export async function getSedi() {
  return apiGet('/sedi/')
}

export async function getSede(id) {
  return apiGet(`/sedi/${id}`)
}

/** @param {{ nome, indirizzo, citta, capienza_massima? }} payload */
export async function creaSede(payload) {
  return apiPost('/sedi/', payload)
}

/** @param {number} id @param {{ nome?, indirizzo?, citta?, capienza_massima? }} payload */
export async function modificaSede(id, payload) {
  return apiPatch(`/sedi/${id}`, payload)
}